push-sender/src/main/java/org/jboss/aerogear/unifiedpush/message/NotificationDispatcher.java

```diff
@@ -62,15 +65,42 @@ public void sendMessagesToPushNetwork(@Observes @Dequeue MessageHolderWithTokens
         final Collection<String> deviceTokens = msg.getDeviceTokens();
 
         logger.info("Received UnifiedPushMessage from JMS queue, will now trigger the Push Notification delivery for the %s variant ({})", variant.getType().getTypeName(), variant.getVariantID());
+        if (sanityCheck(variant) == false) {
+            logger.error("Provided variant has invalid credentials. No push messages are sent to network");
```

Comment:
I like this.

However, I think it would be nice, if that is checked only for iOS, and in teh `sendPushMessage()` that would allow us to also invoke the `onError` on the call back, to show what's going on, in the overview of the push messages send page